torch/utils/ffi/cpp_build.py

```diff
@@ -0,0 +1,18 @@
+import inspect
+import os.path
+
+from setuptools.command.build_ext import build_ext
+
+class BuildExtension(build_ext):
+    """A custom build extension for adding compiler-specific options."""
+    def build_extensions(self):
+        for extension in self.extensions:
+            extension.extra_compile_args = ['-std=c++11']
+        build_ext.build_extensions(self)
+
+
+def include_paths():
+    # https://stackoverflow.com/questions/2632199/how-do-i-get-the-path-of-the-current-executed-file-in-python
+    here = os.path.abspath(inspect.getsourcefile(lambda: 0))
```

Comment:
can't you just use `__file__`?

Reply:
I [read]( https://stackoverflow.com/questions/2632199/how-do-i-get-the-path-of-the-current-executed-file-in-python) that it's less portable/available than this method, though I guess since we're not dealing with an interpreter or anything but a proper Python module that should work. I'll use `__file__` then.